packages/block-editor/src/components/inserter/test/reusable-blocks-tab.js

```diff
@@ -81,36 +49,25 @@ describe( 'InserterMenu', () => {
 			categories,
 			collections,
 		] );
-		const { container } = render(
-			<InserterBlockList filterValue="random" />
-		);
-		const visibleBlocks = container.querySelector(
-			'.block-editor-block-types-list__item'
-		);
 
-		expect( visibleBlocks ).toBe( null );
+		render( <ReusableBlocksTab filterValue="random" /> );
+
+		expect( screen.queryByRole( 'option' ) ).not.toBeInTheDocument();
 	} );
 
 	it( 'should list reusable blocks', () => {
-		const container = initializeAllClosedMenuState();
-		const blocks = container.querySelectorAll(
-			'.block-editor-block-types-list__item-title'
-		);
+		render( <ReusableBlocksTab /> );
 
-		expect( blocks ).toHaveLength( 1 );
-		expect( blocks[ 0 ] ).toHaveTextContent( 'My reusable block' );
+		expect(
+			screen.getByRole( 'option', { name: 'My reusable block' } )
+		).toBeVisible();
 	} );
 
 	it( 'should trim whitespace of search terms', () => {
-		const { container } = render(
-			<InserterBlockList filterValue=" my reusable" />
-		);
-
-		const blocks = container.querySelectorAll(
-			'.block-editor-block-types-list__item-title'
-		);
+		render( <ReusableBlocksTab filterValue=" my reusable" /> );
 
-		expect( blocks ).toHaveLength( 1 );
-		expect( blocks[ 0 ] ).toHaveTextContent( 'My reusable block' );
+		expect(
```

Comment:
Previously, the test was checking that the blocks array had exactly 1 item, should we add that test back? (same for other tests that were implementing similar checks)

Reply:
I don't think we should add that test back, because:

* In all those instances that we change here, the assertions will fail if we have 0 matching elements, or more than 1 matching element.
* If we keep these as-they-are, it would introduce violations of the [`prefer-in-document`](https://github.com/testing-library/eslint-plugin-jest-dom/blob/main/docs/rules/prefer-in-document.md) ESLint rule, which is already enabled throughout the codebase.

That being said, I think it's safe to go with what we're suggesting in this PR. WDYT?

Reply:
Sounds good to me. I had missed the nuance that `getBy` functions expect _exactly one_ matching element

Thank you for the context!